### PR TITLE
Multicast packets forwarding POC

### PR DESCRIPTION
**Pull request by baltrus1**
This is a PoC for multicast communication.
It includes multicast <-> multicast communication (mdns like for eg) as well as support for multicast -> unicast response (SSDP like for eg).

P.S.  Since this is a multicast there are a number of architectural and code optimizations not done. There are also a few tests that might be missing from here.

### :ballot_box_with_check: Definition of Done checklist
- [ ] Commit history is clean ([requirements](../blob/main/docs/git_commit_messages_requirements.md))
- [x] README.md is updated
- [x] changelog.md is updated
- [x] Functionality is covered by unit or integration tests

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -2907,6 +2922,15 @@ dependencies = [
  "no-std-net",
 ]
 
+[[package]]
```
Comment (packgron) [inline Cargo.lock:2925]: We introduced a second version of same lib, can you make sure we reuse version by adding to workspace deps?

File: crates/telio-starcast/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+[dependencies]
+tokio = { workspace = true, features = ["rt", "net", "sync", "macros"] }
+tracing.workspace = true
+boringtun.workspace = true
+x25519-dalek.workspace = true
+base64.workspace = true
+ipnetwork.workspace = true
+pnet_packet.workspace = true
+pnet_transport = "0.34.0"
+etherparse = "0.14.2"
```
Comment (packgron) [inline crates/telio-starcast/Cargo.toml]: why is this and pnet_packet both needed ?

Comment (packgron) [reply crates/telio-starcast/Cargo.toml]: Theu are doing same things

File: crates/telio-starcast/src/multicast_peer.rs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+pub struct MulticasterIp {
+    pub ipv4: Ipv4Addr,
+    pub ipv6: Ipv6Addr,
+}
+
+impl MulticasterPeer {
+    pub async fn new(
+        telio_wg_port: u16,
+        public_key: &PublicKey,
+        wg: Arc<DynamicWg>,
```
Comment (packgron) [inline crates/telio-starcast/src/multicast_peer.rs:32]: Should probably be Arc<dyn Wg>, and why is port needed if we can fetch it from wg?

File: Cargo.lock (hunk 1 above)
Comment (Hasan6979) [reply Cargo.lock]: Fixed!

File: crates/telio-starcast/src/multicast_peer.rs (hunk 3 above)
Comment (Hasan6979) [reply crates/telio-starcast/src/multicast_peer.rs:32]: What should be Arc ? 
Port isn't needed. leftover.

File: crates/telio-starcast/src/multicaster.rs [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+        if let Some(handle) = &self.transport_to_local_task {
+            handle.abort();
+        }
+        if let Some(handle) = &self.tun_to_transport_task {
+            handle.abort();
+        }
+    }
+
+    pub async fn tun_to_transport(
+        read_chan: Arc<RwLock<Chan<Vec<u8>>>>,
```
Comment (packgron) [inline crates/telio-starcast/src/multicaster.rs:107]: This is a heavy anti patten, you should never have channel in RwLock, this causes often leads to deadlocks.

**Pull request closed without merging**